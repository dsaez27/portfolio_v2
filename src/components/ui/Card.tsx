import React from "react";
import styled from "styled-components";

export const Card = ({ title, description, image, link, technologies }) => {
    return (
        <>
            <CardContainer>
                <CardImage src={image} alt={title} />
                <CardContent>
                    <CardTitle>{title}</CardTitle>
                    <CardDescription>{description}</CardDescription>
                    <CardTechnologies>
                        {technologies.map((technology, index) => {
                            return (
                                <CardTechnologyItem key={index}>
                                    {technology}
                                </CardTechnologyItem>
                            );
                        })}
                    </CardTechnologies>
                </CardContent>
            </CardContainer>
        </>
    );
};

const CardContainer = styled.div`
    border: 0px solid aqua;
    cursor: pointer;
    position: relative;
    height: 12rem;
    overflow: hidden;
    border-radius: 3px;
    flex: 1;
    min-width: 260px;
    box-shadow: 0 0 1rem rgb(0 0 0 / 50%);
    -webkit-backface-visibility: hidden;
    -moz-backface-visibility: hidden;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);

    &:before {
        content: "";
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        background: rgba(0, 0, 0, 0)
            linear-gradient(
                to bottom,
                rgba(0, 0, 0, 0) 30%,
                rgb(0 0 0 / 30%) 70%
            );
        z-index: 0;
    }

    &:hover {
        img {
            transform: scale(1.1);
        }
    }
`;

const CardImage = styled.img`
    width: 100%;
    height: 100%;
    display: block;
    object-fit: cover;
    transition: transform 3s ease;
    -webkit-backface-visibility: hidden;
    backface-visibility: hidden;
    position: absolute;
    z-index: -1;
`;

const CardContent = styled.div`
    width: 100%;
    height: 100%;
    padding: 0.5rem;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    * {
        z-index: 1;
    }
`;

const CardTitle = styled.h3`
    font-size: 1.5rem;
    color: #fff;
    margin: 0.5rem 0;
    font-weight: 300;
`;

const CardDescription = styled.p`
    font-size: 0.8rem;
    color: #fff;
    padding: 0;
    font-weight: 300;
`;

const CardTechnologies = styled.div`
    padding: 0;
`;

const CardTechnologyItem = styled.span`
    display: inline-block;
    margin: 0 0.5rem 0 0;
    padding: 0.2rem 0.3rem;
    border-radius: 3px;
    background: #f44653;
    color: #fff;
    font-size: 0.7rem;
`;
