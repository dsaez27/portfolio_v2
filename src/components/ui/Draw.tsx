import React from "react";
import { Fade } from "react-awesome-reveal";
import styled from "styled-components";
import { helperImages } from "../../helpers/helperImages";

export const Draw = () => {
    return (
        <>
            <DrawContainer>
                <Fade direction="bottom" duration="2000" triggerOnce={true}>
                    <Moon src={helperImages("./moon.webp").default} alt="moon" />
                </Fade>

                <Fade direction="left" duration="2000" triggerOnce={true}>
                    <Boy src={helperImages("./boy.webp").default} alt="boy" />
                </Fade>
            </DrawContainer>
        </>
    );
};

const DrawContainer = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: center;
    align-items: center;
`;

const Moon = styled.img`
    transform: translate(100px, 0px);
    width: 120px;
    height: 120px;

    @media (max-width: 768px) {
        transform: translate(120px, 0px);
    }
`;

const Boy = styled.img`
    transform: translate(-100px, 50px);
    width: 100px;
    height: 300px;

    @media (max-width: 768px) {
        transform: translate(-100px, 0px);
    }
`;
