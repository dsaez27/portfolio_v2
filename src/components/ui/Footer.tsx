import { darken } from "@/helpers";

export const Footer = () => {
    return (
        <footer className="flex w-full flex-row items-center justify-center py-3 bg-darken">
            <span className="text-sm font-bold text-white">Made with ❤️ by Daniel Sáez</span>
        </footer>
    );
};
