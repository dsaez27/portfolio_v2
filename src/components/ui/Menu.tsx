import { useState } from "react";
import { createPortal } from "react-dom";
import { Link } from "react-scroll";
import { CSSTransition } from "react-transition-group";

export const Menu = () => {
    const [isOpen, setIsOpen] = useState(false);

    const genericHamburgerLine = `h-0.5 w-6 my-1 rounded-lg transition ease
    transform duration-300 bg-white`;

    return (
        <section className="sm:hidden ">
            <button
                className="group flex flex-col items-center justify-center"
                onClick={() => setIsOpen(!isOpen)}
            >
                <div
                    className={`${genericHamburgerLine} ${
                        isOpen
                            ? "translate-y-2 rotate-45 group-hover:opacity-100 "
                            : "group-hover:opacity-100"
                    }`}
                />
                <div
                    className={`${genericHamburgerLine} ${
                        isOpen ? "opacity-0" : "group-hover:opacity-100"
                    }`}
                />
                <div
                    className={`${genericHamburgerLine} ${
                        isOpen
                            ? "-translate-y-3 -rotate-45 group-hover:opacity-100"
                            : "group-hover:opacity-100"
                    }`}
                />
            </button>
            {isOpen &&
                createPortal(
                    <CSSTransition
                        in={isOpen}
                        timeout={300}
                        classNames="menu-mobile"
                        unmountOnExit
                    >
                        <div className="fixed left-0 top-0 z-10 flex h-screen w-full flex-col items-center justify-evenly bg-darken">
                            <ul className="flex min-h-[250px] flex-col items-center justify-between uppercase text-white">
                                <li className="my-8 cursor-pointer pr-8 transition">
                                    <Link
                                        to="home"
                                        spy={true}
                                        smooth={true}
                                        offset={-70}
                                        duration={500}
                                        onClick={() => setIsOpen(false)}
                                        className="navbar-underline-animation cursor-pointer text-white"
                                    >
                                        Inicio
                                    </Link>
                                </li>
                                <li className="my-8 cursor-pointer pr-8 transition">
                                    <Link
                                        to="about"
                                        spy={true}
                                        smooth={true}
                                        offset={-70}
                                        duration={500}
                                        onClick={() => setIsOpen(false)}
                                        className="navbar-underline-animation cursor-pointer text-white"
                                    >
                                        Sobre MÃ­
                                    </Link>
                                </li>
                                <li className="my-8 cursor-pointer pr-8 transition">
                                    <Link
                                        to="skills"
                                        spy={true}
                                        smooth={true}
                                        offset={-70}
                                        duration={500}
                                        onClick={() => setIsOpen(false)}
                                        className="navbar-underline-animation cursor-pointer text-white"
                                    >
                                        Skills
                                    </Link>
                                </li>
                                <li className="my-8 cursor-pointer pr-8 transition">
                                    <Link
                                        to="projects"
                                        spy={true}
                                        smooth={true}
                                        offset={-70}
                                        duration={500}
                                        onClick={() => setIsOpen(false)}
                                        className="navbar-underline-animation cursor-pointer text-white"
                                    >
                                        Proyectos
                                    </Link>
                                </li>
                                <li className="my-8 cursor-pointer pr-8 transition">
                                    <Link
                                        to="contact"
                                        spy={true}
                                        smooth={true}
                                        offset={-70}
                                        duration={500}
                                        onClick={() => setIsOpen(false)}
                                        className="navbar-underline-animation cursor-pointer text-white"
                                    >
                                        Contacto
                                    </Link>
                                </li>
                            </ul>
                        </div>
                    </CSSTransition>,
                    document.querySelector("#nav") as Element
                )}
        </section>
    );
};
