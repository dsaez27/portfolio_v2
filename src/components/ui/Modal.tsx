import React from 'react';
import styled from 'styled-components';

export const Modal = ({
    children,
    estado,
    cambiarEstado,
    titulo,
    mostrarHeader,
    mostrarOverlay,
    posicionModal,
    padding
}) => {
    return (
        <>
            {estado && (
                <Overlay
                    mostrarOverlay={mostrarOverlay}
                    positionModal={posicionModal}
                >
                    <ContenedorModal padding={padding}>
                        <EncabezadoModal>
                            {mostrarHeader && <h3>{titulo}</h3>}
                        </EncabezadoModal>
                        <ContenedorBotonCerrar>
                            <BotonCerrar onClick={() => cambiarEstado(false)}>
                                <svg
                                    xmlns='http://www.w3.org/2000/svg'
                                    width='16'
                                    height='16'
                                    fill='currentColor'
                                    className='bi bi-x'
                                    viewBox='0 0 16 16'
                                >
                                    <path d='M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z' />
                                </svg>
                            </BotonCerrar>
                        </ContenedorBotonCerrar>
                        {children}
                    </ContenedorModal>
                </Overlay>
            )}
        </>
    );
};

const size = {
    xs: '767px',
    sm: '768px',
};

const device = {
    xs: `(max-width: ${size.xs})`,
    sm: `(min-width: ${size.sm})`,
};

const Overlay = styled.div`
    width: 100vw;
    height: 100vh;
    position: fixed;
    top: 0;
    left: 0;
    background-color: ${(props) =>
        props.mostrarOverlay ? 'rgba(0, 0, 0, 0.5)' : 'rgba(0, 0, 0, 0)'};
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: ${(props) =>
        props.posicionModal ? props.posicionModal : 'center'};

    z-index: 10;
`;

const ContenedorModal = styled.div`
    @media ${device.xs} {
        max-width: 85%;
        max-height: 90vh;
        margin: 1rem 0;
    }
    @media ${device.sm} {
        max-width: 500px;
        max-height: 90vh;
        margin: 1rem 0;
    }
    overflow-y: overlay;
    min-height: 100px;
    background-color: ${(props) =>
        props.background ? props.background : '#212336'};
    position: relative;
    border-radius: 3px;
    box-shadow: 0 7px 29px 0 rgba(100, 100, 111, 0.2);
    padding: ${(props) => (props.padding ? props.padding : '20px')};
    &::-webkit-scrollbar {
        width: 8px;
    }
    &::-webkit-scrollbar-thumb {
        background: #ccc;
        border-radius: 3px;
    }
    &::-webkit-scrollbar-thumb:active {
        background-color: #999999;
    }
    &::-webkit-scrollbar-thumb:hover {
        background: #b3b3b3;
        box-shadow: 0 0 2px 1px rgba(0, 0, 0, 0.2);
    }
    &::-webkit-scrollbar-track {
        background: #e1e1e1;
        border-radius: 3px;
    }
    &::-webkit-scrollbar-track:hover,
    &::-webkit-scrollbar-track:active {
        background: #d4d4d4;
    }
`;

const EncabezadoModal = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 0px;
    padding-bottom: 0px;
    font-family: 'Roboto', sans-serif;
    h3 {
        font-weight: 600;
        font-size: 16px;
        color: #1766dc;
`;

const BotonCerrar = styled.div`
    width: 30px;
    height: 30px;
    border: none;
    background: none;
    cursor: pointer;
    transition: all ease 0.3s;
    border-radius: 3px;
    color: #1766dc;
    &:hover {
        background: rgba(10, 10, 10, 0.6);
    }
    svg {
        width: 100%;
        height: 100%;
    }
`;

const ContenedorBotonCerrar = styled.div`
    position: absolute;
    top: 0;
    right: 0;
    padding: 20px;
`;
