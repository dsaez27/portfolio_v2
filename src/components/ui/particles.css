.tsparticles-canvas-el {
    position: absolute !important;
    width: 100% !important;
    height: 100vh !important;
    z-index: 0 !important;
    padding: 0% !important;
}

@media screen and (max-width: 768px) {
    .tsparticles-canvas-el {
        height: 200vh !important;
    }
}
