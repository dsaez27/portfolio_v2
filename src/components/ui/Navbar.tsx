import React, { useCallback, useEffect, useRef, useState } from "react";
import { Link } from "react-scroll";
import { Menu } from "./Menu";

const Navbar = () => {
    const [bgOpacity, setBgOpacity] = useState(0);
    const bgOpacityRef = useRef(bgOpacity);

    const handleScroll = useCallback(() => {
        const scrollPercentage = Math.floor(
            (window.scrollY / window.innerHeight) * 100
        );
        const rounded = Math.round(scrollPercentage / 5) * 5;
        rounded <= 75 ? setBgOpacity(rounded) : setBgOpacity(0);
    }, []);

    useEffect(() => {
        bgOpacityRef.current = bgOpacity;
    }, [bgOpacity]);

    useEffect(() => {
        document.addEventListener("scroll", handleScroll);
        return () => {
            document.removeEventListener("scroll", handleScroll);
        };
    }, [handleScroll]);

    return (
        <nav
            className={`navbar fixed z-50 w-full bg-darken py-2.5 transition duration-1000 selection:ease-in-out max-sm:px-2 sm:px-4`}
            style={
                {
                    "--tw-bg-opacity": bgOpacityRef.current / 100,
                } as React.CSSProperties
            }
        >
            <div className="container mx-auto flex max-w-4xl flex-wrap items-center justify-between">
                <h2 className="cursor-default self-center whitespace-nowrap text-xl font-bold text-white first-letter:text-primary">
                    DS√ÅEZ
                </h2>
                <ul className="flex gap-5 font-medium max-sm:hidden">
                    <li>
                        <Link
                            to="home"
                            spy={true}
                            smooth={true}
                            offset={-70}
                            duration={500}
                            className="navbar-underline-animation cursor-pointer text-white"
                        >
                            Inicio
                        </Link>
                    </li>
                    <li>
                        <Link
                            to="skills"
                            spy={true}
                            smooth={true}
                            offset={-70}
                            duration={500}
                            className="navbar-underline-animation cursor-pointer text-white"
                        >
                            Skills
                        </Link>
                    </li>
                    <li>
                        <Link
                            to="projects"
                            spy={true}
                            smooth={true}
                            offset={-70}
                            duration={500}
                            className="navbar-underline-animation cursor-pointer text-white"
                        >
                            Proyectos
                        </Link>
                    </li>
                    <li>
                        <Link
                            to="contact"
                            spy={true}
                            smooth={true}
                            offset={-70}
                            duration={500}
                            className="navbar-underline-animation cursor-pointer text-white"
                        >
                            Contacto
                        </Link>
                    </li>
                </ul>
                <Menu />
            </div>
        </nav>
    );
};

export default Navbar;
