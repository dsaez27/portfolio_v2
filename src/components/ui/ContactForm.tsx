import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import * as yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup';
import { Modal } from './Modal';
import styled from 'styled-components';
import { helperImages } from '../../helpers/helperImages';
import { Rotate } from 'react-awesome-reveal';
import axios from 'axios';

const schema = yup.object().shape({
    name: yup.string().required('Nombre es requerido'),
    email: yup
        .string()
        .email('Por favor ingrese un email válido')
        .required('Email es requerido'),
    message: yup
        .string()
        .required('Mensaje es requerido')
        .min(10, 'Mínimo 10 caracteres')
        .max(100, 'Máximo 100 caracteres'),
});

export const ContactForm = () => {
    const [estadoModal, setEstadoModal] = useState(false);

    const {
        register,
        handleSubmit,
        formState: { errors },
        reset,
    } = useForm({
        resolver: yupResolver(schema),
    });

    const onSubmit = async (data) => {
        axios
            .post('/contact', data)
            .then((res) => {
                setEstadoModal(true);
                reset();
            })
            .catch((err) => {
                console.log(err);
            });
    };

    return (
        <>
            <form onSubmit={handleSubmit(onSubmit)} id='form-validation'>
                <label htmlFor='name'>Nombre</label>
                <input
                    id='name'
                    type='text'
                    placeholder='Nombre y apellido'
                    {...register('name')}
                />
                <p>{errors.name?.message}</p>
                <label htmlFor='email'>Email</label>
                <input
                    id='email'
                    type='email'
                    placeholder='Email'
                    {...register('email', {
                        required: true,
                    })}
                />
                <p>{errors.email?.message}</p>
                <label htmlFor='message'>Mensaje</label>
                <textarea
                    id='message'
                    placeholder='Mensaje'
                    rows='10'
                    cols='50'
                    {...register('message')}
                />
                <p>{errors.message?.message}</p>
                <button type='submit'>Enviar</button>
            </form>

            <Modal
                estado={estadoModal}
                cambiarEstado={setEstadoModal}
                mostrarHeader={true}
                mostrarOverlay={true}
                posicionModal={'center'}
                padding={'4rem 2rem 1rem'}
            >
                <Contenido>
                    <Rotate>
                        <img
                            src={helperImages('./check.webp').default}
                            alt='check'
                        />
                    </Rotate>
                    <Titulo>Éxito</Titulo>
                    <p>Su mensaje ha sido enviado correctamente</p>
                    <Boton onClick={() => setEstadoModal(false)}>Aceptar</Boton>
                </Contenido>
            </Modal>
        </>
    );
};

const Boton = styled.div`
    display: block;
    padding: 10px 30px;
    border-radius: 100px;
    border: none;
    background: #1766dc;
    cursor: pointer;
    font-weight: 500;
    transition: 0.3s ease all;
    color: #fff;
    &:hover {
        background: #0066ff;
    }
`;

const Contenido = styled.div`
    text-align: center;
    p {
        font-size: 1rem;
        margin-bottom: 20px;
    }
`;

const Titulo = styled.h1`
    color: #fff;
    font-size: 2rem;
    font-weight: 500;
    margin-top: 20px;
`;
