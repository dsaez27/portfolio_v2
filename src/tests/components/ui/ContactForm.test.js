import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import { ContactForm } from '../../../components/ui/ContactForm';
import { Modal } from './../../../components/ui/Modal';
import { mount } from 'enzyme';

describe('Pruebas en <ContactForm />', () => {
    beforeEach(() => {
        // IntersectionObserver isn't available in test environment
        const mockIntersectionObserver = jest.fn();
        mockIntersectionObserver.mockReturnValue({
            observe: () => null,
            unobserve: () => null,
            disconnect: () => null,
        });
        window.IntersectionObserver = mockIntersectionObserver;
    });

    test('Debe de mostrarse el formulario', () => {
        const wrapper = render(<ContactForm />);
        expect(wrapper).toMatchSnapshot();
    });

    test('mostrar error en input nombre', async () => {
        const { findByLabelText, findByText, findByRole } = render(
            <ContactForm />
        );

        const inputName = await findByLabelText('Nombre');
        await act(async () => {
            fireEvent.change(inputName, { target: { value: '' } });
            fireEvent.submit(await findByRole('button'));
        });

        const error = await findByText('Nombre es requerido');
        expect(error).toBeInTheDocument();
    });

    test('mostrar error en input email', async () => {
        const { findByLabelText, findByText, findByRole } = render(
            <ContactForm />
        );

        const inputEmail = await findByLabelText('Email');

        await act(async () => {
            fireEvent.change(inputEmail, { target: { value: '' } });
            fireEvent.submit(await findByRole('button'));
        });

        const error = await findByText('Email es requerido');

        expect(error).toBeInTheDocument();
    });

    test('Mostrar error al ingresar email inválido', async () => {
        const { findByLabelText, findByText, findByRole } = render(
            <ContactForm />
        );
        const inputEmail = await findByLabelText('Email');
        await act(async () => {
            fireEvent.change(inputEmail, { target: { value: 'hola' } });
            fireEvent.submit(await findByRole('button'));
        });
        const error = await findByText('Por favor ingrese un email válido');
        expect(error).toBeInTheDocument();
    });
    test('mostrar error en input mensaje', async () => {
        const { findByLabelText, findByText, findByRole } = render(
            <ContactForm />
        );
        const inputMessage = await findByLabelText('Mensaje');
        await act(async () => {
            fireEvent.change(inputMessage, { target: { value: '' } });
            fireEvent.submit(await findByRole('button'));
        });
        const error = await findByText('Mensaje es requerido');
        expect(error).toBeInTheDocument();
    });

    test('Debe de enviar el formulario', async () => {
        const { findByLabelText, findByRole } = render(<ContactForm />);
        const inputName = await findByLabelText('Nombre');
        const inputEmail = await findByLabelText('Email');
        const inputMessage = await findByLabelText('Mensaje');
        await act(async () => {
            fireEvent.change(inputName, { target: { value: 'Juan' } });
            fireEvent.change(inputEmail, {
                target: { value: 'dssh27@gmail.com' },
            });
            fireEvent.change(inputMessage, { target: { value: 'Hola' } });
            fireEvent.submit(await findByRole('button'));
        });
        expect(inputName.value).toBe('Juan');
        expect(inputEmail.value).toBe('dssh27@gmail.com');
        expect(inputMessage.value).toBe('Hola');
    });

    test('Debe de mostrar modal de exito', async () => {
        const { findByLabelText, findByRole, findAllByText } = render(
            <ContactForm />
        );
        const { rerender } = render(<Modal cambiarEstado={false} />);

        const inputName = await findByLabelText('Nombre');
        const inputEmail = await findByLabelText('Email');
        const inputMessage = await findByLabelText('Mensaje');
        await act(async () => {
            fireEvent.change(inputName, { target: { value: 'Juan' } });
            fireEvent.change(inputEmail, {
                target: { value: 'dssh27@gmail.com' },
            });
            fireEvent.change(inputMessage, { target: { value: '0123456789' } });
            fireEvent.submit(await findByRole('button'));
        });

        rerender(<Modal cambiarEstado={true} />);

        expect(
            await findAllByText('Su mensaje ha sido enviado correctamente')
        ).toBeTruthy();
    });
});
