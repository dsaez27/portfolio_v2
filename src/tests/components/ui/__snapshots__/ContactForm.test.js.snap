// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ContactForm /> Debe de mostrarse el formulario 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        id="form-validation"
      >
        <label
          for="name"
        >
          Nombre
        </label>
        <input
          id="name"
          name="name"
          placeholder="Nombre y apellido"
          type="text"
        />
        <p />
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          placeholder="Email"
          type="email"
        />
        <p />
        <label
          for="message"
        >
          Mensaje
        </label>
        <textarea
          cols="50"
          id="message"
          name="message"
          placeholder="Mensaje"
          rows="10"
        />
        <p />
        <button
          type="submit"
        >
          Enviar
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      id="form-validation"
    >
      <label
        for="name"
      >
        Nombre
      </label>
      <input
        id="name"
        name="name"
        placeholder="Nombre y apellido"
        type="text"
      />
      <p />
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        placeholder="Email"
        type="email"
      />
      <p />
      <label
        for="message"
      >
        Mensaje
      </label>
      <textarea
        cols="50"
        id="message"
        name="message"
        placeholder="Mensaje"
        rows="10"
      />
      <p />
      <button
        type="submit"
      >
        Enviar
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
